### DISCUSSION TITLE
getIcon result in a 'The source image could not be decoded' error.

### DISCUSSION THREAD

**Discussion by pmcneves** [visgl/deck.gl#8007] [Q&A]
Hello everyone.

I have a playground with this IconLayer, described below:
```
new IconLayer({
      id: 'icon-layer',
      getPosition: d => d.location.coordinates,
      data: iconData,
      getIcon: d => ({
          url: 'http://localhost:3001/svg',
          width: 15,
          height: 15
        });
      )
```
      
This is my minimal express endpoint:
```
app.get('/svg', (req, res) => {  
  fs.readFile('./icon.svg', 'utf8', (err, data) => {  
    if (err) {  
      res.status(500).send('Internal Server Error');  
    } else {  
      res.setHeader('Content-Type', 'image/svg+xml');  
      res.send(data);  
    }
  });
});

```

Using my server, I get the error ''The source image could not be decoded'. However, if my IconLayer getIcon function has the url `https://www.svgrepo.com/show/528833/add-square.svg`, it works and the icons are rendered.. It's the same icon, just different origin. How can I get around this? Thanks in advance :)

**Comment by ElfarriBadr** [comment]
I am having the same issue, is there any work around here?
